tools/get_features.py

```diff
@@ -151,15 +159,17 @@ def get_features(
             raise ValueError(f"No data found for source ids {source_ids}")
 
         df_temp = pd.DataFrame(source_data)
-        df_temp = df_temp.astype(dtype=dtype_dict)
+        if projection == []:
+            df_temp = df_temp.astype(dtype=dtype_dict)
         df_collection += [df_temp]
         try:
             dmdt_temp = np.expand_dims(
                 np.array([d for d in df_temp['dmdt'].values]), axis=-1
             )
         except Exception as e:
-            print("Error", e)
-            print(df_temp)
+            if "dmdt" in projection:
```

Comment:
Should there be an else here?

Reply:
@mcoughlin I pushed a slight change to this code, but I don't think there is need for an else here. The if statement prints any dmdt-related exception if the user specifically requests dmdt or is using the default projection. Otherwise, the exception can be suppressed, since dmdt will not be among the features returned by the query.

Reply:
Sounds good :)